internal/incident/incident.go

```diff
@@ -287,9 +278,9 @@ func (i *Incident) evaluateRules(eventID int64, causedBy types.Int) (types.Int,
 			}
 			insertedID, err := i.AddRuleMatchedHistory(r, history)
 			if err != nil {
-				i.logger.Errorln(err)
+				i.logger.Errorw("Failed to add incident rule matched history", zap.String("rule", r.Name), zap.Error(err))
 
-				return types.Int{}, err
+				return types.Int{}, errors.New("failed to add incident rule matched history")
 			}
```

Comment:
Same here, that function actually does more than just write history (even despite its name).